org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/structure/constraints/SuperTypeRefactoringProcessor.java

```diff
@@ -433,12 +431,12 @@ protected final String createTypeSource(final ICompilationUnit copy, final IType
 				source= buffer.toString();
 			}
 			final IDocument document= new Document(source);
-			final TextEdit edit= CodeFormatterUtil.format2(CodeFormatter.K_COMPILATION_UNIT, source, 0, delimiter, FormatterProfileManager.getProjectSettings(copy.getJavaProject()));
+			final TextEdit edit= CodeFormatterUtil.format2(CodeFormatter.K_COMPILATION_UNIT, source, 0, delimiter, copy.getJavaProject().getOptions(true));
```

Comment:
This change omits some logic that is found in FormatterProfileManager.getProjectSettings() which looks at the preferences.  Admittedly, there is logic there that uses JDT UI classes but some way around it should be found.

Reply:
Yeah, we need to take a closer look here. I think we might have migrated some classes in the past and we probably had to deal with this, so we just need to determine what we did, or what the best approach is to support the behaviour.

Reply:
I just move the implementation of method `getProjectSettings()` to a new type FormatterProfileManagerCore via https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/344/commits/03bfc3ea777b8248b811f32fa4df68364d837aec, since moving whole `FormatterProfileManager` will cause a lot of other changes. Now we can reach the logics there.

Reply:
Looks good.